src/tracker/long_tracker.jl

```diff
@@ -98,73 +105,69 @@ Find matches for floes in `tracked` from floes in  `candidate_props`.
+    matches = []
+    for floe1 in eachrow(tracked), floe2 in eachrow(candidate_props)
+        Δt = get_dt(floe1, floe2)
+        ratios, conditions, dist = compute_ratios_conditions(
+            floe1, floe2, Δt, condition_thresholds
+        )
 
-                if callmatchcorr(conditions)
-                    @debug "Getting mismatch and correlation for floe 1:$r and floe 2:$s"
-                    (area_mismatch, corr) = matchcorr(
-                        props1.mask[r], props2.mask[s], Δt; mc_thresholds.comp...
-                    )
-
-                    if isfloegoodmatch(
-                        conditions, mc_thresholds.goodness, area_mismatch, corr
-                    )
-                        @debug "** Found a good match for floe 1:$r => 2:$s"
-                        appendrows!(
-                            matching_floes,
-                            props2[s, :],
-                            (ratios..., area_mismatch, corr),
-                            s,
-                            dist,
-                        )
-                        @debug "Matching floes" matching_floes
-                    end
-                end
-            end # of s for loop
-
-            # 2. Find the best match for floe r
-            @debug "Finding best match for floe 1:$r"
-            best_match_idx = getidxmostminimumeverything(matching_floes.ratios)
-            @debug "Best match index for floe 1:$r: $best_match_idx"
-            if isnotnan(best_match_idx)
-                bestmatchdata = getbestmatchdata(best_match_idx, r, props1, matching_floes) # might be copying data unnecessarily
-                addmatch!(matched_pairs, bestmatchdata)
-                @debug "Matched pairs" matched_pairs
+        if callmatchcorr(conditions)
+            (area_mismatch, corr) = matchcorr(
+                floe1.mask, floe2.mask, Δt; mc_thresholds.comp...
+            )
+            if isfloegoodmatch(conditions, mc_thresholds.goodness, area_mismatch, corr)
+                @debug "** Found a good match for ", floe1.uuid, "<=", floe2.uuid
+                push!(
+                    matches,
+                    (;
+                        Δt,
+                        measures=(; ratios..., area_mismatch, corr),
+                        conditions,
+                        dist,
+                        floe1,
+                        floe2,
+                    ),
+                )
             end
-        end # of for r = 1:nrow(props1)
+        end
+    end
 
-        # exit while loop if there are no more floes to match
-        @debug "Matched pairs" matched_pairs
-        isempty(matched_pairs) && break
+    remaining_matches_df = DataFrame(matches)
+    best_matches = []
 
-        #= Resolve collisions:
-        Are there floes in day k+1 paired with more than one
-        floe in day k? If so, keep the best matching pair and remove all others. =#
+    for floe2 in eachrow(candidate_props)  # leave at most one match for each 
```

Comment:
Move sorting into here

Reply:
Make this function (less the `push!`) into a configurable NamedTrackingAlgorithm.